Assets/Source/Components/Pathing/ProjectilePathfinderComponent.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Assets.Source.AStar;
+using Assets.Source.Components.Projectile.Base;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace Assets.Source.Components.Pathing
+{
+    /// <summary>
+    /// The pathfinder component relies on an existing navigation mesh to find its way around obstacles
+    /// </summary>
+    public abstract class ProjectilePathfinderComponent : ProjectileComponentBase
```

Comment:
Is the idea behind this to have bullets or missiles that move towards the target and can avoid obstacles?

Reply:
well, because enemies are currently projectiles too, i had to do it this way for now.